aicsimageio/tests/test_dimensions.py

```diff
@@ -0,0 +1,18 @@
+#!/usr/bin/env python
+# -*- coding: utf-8 -*-
+
+import pytest
+
+from aicsimageio.dimensions import Dimensions
+
+
+def test_dimensions_getitem() -> None:
+    dims = Dimensions("TCZYX", (1, 4, 75, 624, 924))
+    assert dims["T"] == 1
+    assert dims["T", "C"] == (1, 4)
+
+    # out of order indexing
+    assert dims["C", "T", "Y"] == (4, 1, 624)
+
+    with pytest.raises(TypeError):
+        dims[0]  # type: ignore
```

Comment:
Hah. I just think it's funny that you are literally saying this is a TypeError and mypy isn't smart enough to catch it.